types/vfile-location-tests.ts

```diff
@@ -0,0 +1,20 @@
+import * as vfileLocation from 'vfile-location'
+import * as vfile from 'vfile'
+
+// $ExpectError
+vfileLocation()
+
+vfileLocation(vfile('example'))
+vfileLocation('example')
+
+const location = vfileLocation('example')
+
+location.toOffset({line: 1, column: 1})
+// $ExpectError
+location.toOffset(12)
+// $ExpectError
+location.toOffset({line: 1, column: 1, offset: 1})
```

Comment:
currently the typing with warn/error if an offset is passed to `toOffset`.
This restriction could be loosened if preferred.